version: '3'

services:

  db:
    build:
      context: ./database
    command: --innodb-use-native-aio=0
    container_name: db
    restart: always
    tty: true
    volumes:
      - ./mysql:/var/lib/mysql
      - ./database/init.sql:/docker-entrypoint-initdb.d/init.sql
    environment:
      - MYSQL_DATABASE=nodedb
      - MYSQL_ROOT_PASSWORD=root
    networks:
      - redeinterna
    healthcheck:
      test: [ "CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "root", "-p$$MYSQL_ROOT_PASSWORD" ]
      interval: 5s
      timeout: 60s
      retries: 6

  nodeapp:
    build:
      context: ./node
    container_name: nodeapp
    image: ailssonhafemann/nodeapp
    tty: true
    networks:
      - redeinterna
    healthcheck:
      test: [ "CMD","curl", "-f","http://localhost:3000"]
      interval: 5s
      timeout: 30s
      retries: 6
    depends_on:
      db:
        condition: service_healthy

  ngapp:
    build:
      context: ./nginx
    image: ailssonhafemann/ngapp
    container_name: ngapp
    networks:
      - redeinterna
    ports:
      - "8080:80"
    depends_on:
      nodeapp:
        condition: service_healthy

networks:
  redeinterna:
    driver: bridge

volumes:
  mysql:
    driver: local